kernel/include/nano_internal.h

```diff
@@ -82,6 +82,56 @@ static inline unsigned int _Swap(unsigned int key)
 
 #ifdef CONFIG_USERSPACE
 /**
+ * @brief get the maximum number of free regions for memory domain partitions
+ *
+ * @return Max number of free regions
+ */
+extern u8_t _arch_get_max_partitions(void);
+
+/**
+ * @brief Configure MPU memory domain
+ *
+ * This function configures per thread memory domain reprogramming the MPU.
+ * Used when adding the currently running thread to a domain.
+ *
+ * @param thread thread info data structure.
+ */
+extern void _arch_configure_mem_domain(struct k_thread *thread);
+
+/**
+ * @brief Configure MPU memory partition
+ *
+ * This function configure the specific memory partition of a thread. Used
+ * when adding partitions to an active memory domain.
+ *
+ * @param thread thread info data structure.
+ * @param part_index partition index that needs to be configured.
+ */
+extern void _arch_configure_mem_partition(struct k_thread *thread,
+					  u32_t part_index);
```

Comment:
Same complaint: "configure" doesn't really mean anything.   How about _arch_thread_enable_partition() or somesuch instead?

Also seems like this is going to have to validate that part_index, which means that it probably needs to return an error code at runtime.

Reply:
I'm not sure I agree this will need a return value.
These are supervisor only APIs so I think any implementation will simply need to __ASSERT() that its parameters are valid.